src/net/discovery/conf.rs

```diff
@@ -1,9 +1,17 @@
-use crate::net::discovery::{peer, Error, PeerDiscovery};
+use crate::{
+    helpers::HelperIdentity,
+    net::discovery::{peer, Error, PeerDiscovery},
+};
+use std::collections::HashMap;
 
 /// All config value necessary to discover other peer helpers of the MPC ring
-#[cfg_attr(feature = "enable-serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "enable-serde",
+    derive(serde::Deserialize),
+    serde(transparent)
+)]
 pub struct Conf {
-    peers: [peer::Config; 3],
+    peers_map: HashMap<HelperIdentity, peer::Config>,
```

Comment:
is it possible to use the same trick as with role? i.e. implement `Index` for `HelperIdentity` and hold an array here. It may be better for consistency reasons

Reply:
sure. `HelperIdentity` is 1-indexed so gotta subtract 1, but got that done